corehq/apps/linked_domain/tests/test_update_roles.py

```diff
@@ -220,6 +221,25 @@ def test_error_message_indicates_pull(self):
                 ' "Sync & Overwrite" to overwrite and link the matching ones.'):
             update_user_roles(self.domain_link, is_pull=True)
 
+    @patch('corehq.apps.linked_domain.updates.get_brief_apps_in_domain')
+    def test_web_apps_permissions(self, get_brief_apps_in_domain):
+        self._create_user_role(self.upstream_domain, name='Linked & unlinked', permissions=HqPermissions(
+            edit_data=True,
+            edit_reports=True,
+            access_web_apps=False,
+            web_apps_list=['123', '456']
+        ))
+
+        downstream_app = LinkedApplication(
+            _id='789', domain=self.upstream_domain, version=1, name='linked-app', upstream_app_id='123'
+        )
+        get_brief_apps_in_domain.return_value = [downstream_app]
+
+        update_user_roles(self.domain_link)
+
+        role = UserRole.objects.get_by_domain(self.downstream_domain)[0]
+        self.assertListEqual(role.permissions.web_apps_list, [downstream_app.id])
```

Comment:
Ah this clarifies it, thank you.  In my words, it's that a role on an upstream domain with access to an app needs to be propagated to the downstream domain with the app access updated to the corresponding downstream app.  Did I get that right?

Reply:
yep, exactly